File: 11328.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;

	int alpha1[26], alpha2[26];
	for (int i = 0; i < N; i++) {
		for (int j = 0; j < 26; j++)
			alpha1[j] = alpha2[j] = 0;

		string s1, s2; cin >> s1 >> s2;

		if (s1.length() != s2.length()) {
			cout << "Impossible\n";
			continue;
		}
		else {
			for (int j = 0; j < s1.length(); j++) {
				alpha1[s1[j] - 'a']++;
				alpha2[s2[j] - 'a']++;
			}
		}

		bool possible = true;
		for (int j = 0; j < 26; j++) {
			if (alpha1[j] != alpha2[j]) {
				possible = false;
				break;
			}
		}

		if (possible) cout << "Possible\n";
		else cout << "Impossible\n";
	}
}

File: 11328.py
for i in range(int(input())):
    alpha1 = [0] * 26
    alpha2 = [0] * 26

    s1, s2 = input().split()

    if len(s1) != len(s2):
        print("Impossible")
        continue
    else:
        for j in range(len(s1)):
            alpha1[ord(s1[j]) - ord('a')] += 1
            alpha2[ord(s2[j]) - ord('a')] += 1

    if alpha1 == alpha2:
        print("Possible")
    else:
        print("Impossible")
